cmd/poseidon/main.go

```diff
@@ -193,11 +196,16 @@ func initServer() *http.Server {
 
 // shutdownOnOSSignal listens for a signal from the operating system
 // When receiving a signal the server shuts down but waits up to 15 seconds to close remaining connections.
-func shutdownOnOSSignal(server *http.Server) {
+func shutdownOnOSSignal(server *http.Server, ctx context.Context) {
 	// wait for SIGINT
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
-	<-signals
+	select {
+	case <-ctx.Done():
```

Comment:
Just for my understanding: the context will be `Done` when cancelled, i.e., when a panic occurs? Otherwise, Poseidon will run until a signal is received?

And shouldn't we include lines 209-215 before closing the select in line 207 (so that it's clearer what happens there)?

Reply:
> the context will be `Done` when cancelled, i.e., when a panic occurs?

Yes, when our http server stops (due to a panic or an error).

> Otherwise, Poseidon will run until a signal is received?

Yes.

> And shouldn't we include lines 209-215 before closing the select in line 207 (so that it's clearer what happens there)?

Yeah ok :+1: